worker/include/handles/UnixStreamSocket.hpp

```diff
@@ -15,8 +15,14 @@ class UnixStreamSocket
 		uint8_t store[1];
 	};
 
+	enum SocketRole
+	{
+		CONSUMER,
+		PRODUCER
+	};
```

Comment:
We always use `class enum` when possible. Can it be moved to this?

```c++
enum class Role
{
	PRODUCER = 1,
	CONSUMER
};
```

Note that I've also renamed it from `SocketRole` to just `Role`.  Being a `class enum` you need to later reference it as `UnixStreamSocket::Role`.

Reply:
agree. already changed.